### PR TITLE
ci(windows): test using GitHub Actions

### PR DESCRIPTION
**Pull request by myii**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

Built upon the work done by @dafyddj in other formulas:

* https://github.com/saltstack-formulas/openvpn-formula/pull/129
* https://github.com/saltstack-formulas/salt-formula/pull/471

Brings in automated testing for Chocolatey as discussed in #60 (CC: @kartnico).

### Describe the changes you're proposing

Add Windows testing for Chocolatey installations.

### Pillar / config required to test the proposed changes

As provided.

### Debug log showing how the proposed changes work

As seen in the GitHub Actions logs.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [x] Included in Kitchen (i.e. under `state_top`).
- [x] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### Additional context

@dafyddj As mentioned in Slack, I haven't managed to get the Vagrant testing working yet so I'd appreciate it if you have any suggestions for that.

Using this PR as a precursor to providing Windows testing configuration via. the `ssf-formula`.

### CODE REVIEW

**Comment by myii** [comment]
After feedback in the Formulas' working group meeting, tested some modifications in another branch and getting much better performance now:

* https://github.com/myii/packages-formula/runs/972293120?check_suite_focus=true
  - Avoided the Chocolatey installation via. `packages.chocolatey.required.pkgs` -- also allowed an empty `init_environment` in `kitchen.github.yml`.
  - Avoided installing `adobereader`.

@dafyddj mentioned that he would be able to add Chocolatey to the pre-salted image used by `kitchen.vagrant.yml`, so this PR will be checked again locally once that's done.

Also need to add the relevant documentation to `README.rst`.

As for the `debian-10` failure, that's unrelated to this PR.  Appears to be an issue with that particular snap but would need to dig down further, if I can make time.  Otherwise, may have to just disable it in the test pillar.

**Comment by dafyddj** [comment]
@myii uploaded a new Vagrant box version. Do you want to try [v2020.8.198](https://app.vagrantup.com/techneg/boxes/win81x64-pro-salt/versions/2020.8.198)?

**Comment by myii** [comment]
> @myii uploaded a new Vagrant box version. Do you want to try [v2020.8.198](https://app.vagrantup.com/techneg/boxes/win81x64-pro-salt/versions/2020.8.198)?

Just confirming here that the new box works -- thanks.  I've updated the PR accordingly, including removing unnecessary installations to make the Windows testing faster.  So this PR should be ready for merging.  The `debian-10` failure is unrelated.

CC: @dafyddj @javierbertoli.

File: kitchen.github.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  username: kitchen
+  password: Pass@word1
+
+platforms:
+  - name: windows-81-latest-py3
+
+provisioner:
+  salt_install: bootstrap
+  salt_bootstrap_options: -pythonVersion 3
+  init_environment: ""
```
Comment (dafyddj) [inline kitchen.github.yml:16]:
```suggestion
```

Comment (dafyddj) [reply kitchen.github.yml:16]: Would suggest just removing the unneeded lines.

File: kitchen.vagrant.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  name: vagrant
+
+platforms:
+  - name: windows-81-latest-py3
+    driver:
+      box: techneg/win81x64-pro-salt
+      gui: false
+      linked_clone: true
+    provisioner:
+      init_environment: ""
```
Comment (dafyddj) [inline kitchen.vagrant.yml:14]:
```suggestion
```

Comment (dafyddj) [reply kitchen.vagrant.yml:14]: And here.

**Comment by dafyddj** [comment]
Would also suggest changing the platform name to just `windows` for consistency.

**Comment by myii** [comment]
Updated and finalised using https://github.com/myii/ssf-formula/pull/304.

**Comment by myii** [comment]
Thanks for the review, @dafyddj.

**Pull request merged**